lib/Dialect/Ibis/Transforms/IbisTunneling.cpp

```diff
@@ -121,8 +121,14 @@ void Tunneler::genPortNames(llvm::MapVector<StringAttr, PortInfo> &pis) {
       },
       "_");
 
-  for (auto &[sym, portInfo] : pis)
-    portInfo.portName = rewriter.getStringAttr(pathName + "_" + sym.getValue());
+  for (PortInfo &pi : portInfos) {
+    // Suffix the ports by the intended usage (read/write). This also de-aliases
+    // cases where one both reads and writes from the same input port.
+    std::string suffix =
+        pi.getRequestedDirection() == Direction::Input ? ".wr" : ".rd";
```

Comment:
Will I see these suffixes on module ports in the generated RTL?  If so, then like with the hw eliminate in/out pass, it would be nice if these were a parameter to the pass.

Reply:
As of now, yes. These are used to de-alias multiple accesses to the same port, but as with the other pass, if you know that that's not a situation you're putting yourself in, then the suffixes can be omitted.